hashicorp/terraform-provider-aws
# Add ownership-verification-cert. to API Gateway

- add `ownership_verification_certificate_arn` to:
  - `aws_apigatewayv2_domain_name.domain_name_configuration`,
  - `aws_api_gateway_domain_name`.
- update documentation to include `ownership_verification_certificate_arn`;
- add separate tests for mTLS with ownership-verification (as it requires both a public cert. for verification plus an imported cert. to trigger the requirement.)
  - the existing test framework provides `testAccAWSAPIGatewayV2DomainNameConfigPublicCert()` and `testAccAWSAPIGatewayV2DomainNameConfigImportedCerts()` for creating certs. However, both create a `aws_acm_certificate.test` resource (i.e. a conflicting name.) As this change effectively requires both (mTLS with an imported ACM cert. but using an ACM-issued cert. for verification), I've opted to use the former and use a separate, explicit config. to do the latter.
  - I've also added entirely separate tests for these rather than adding a `TestStep` to an existing `ParallelTest` as, largely due to the above, this didn't immediately seem to fit the pattern of the existing tests.
- according to AWS' [docs.](https://docs.aws.amazon.com/apigateway/latest/developerguide/rest-api-mutual-tls.html#rest-api-mutual-tls-non-acm-certs), this field is needed for imported certs. or when using ACM PCA—the tests verify the former but due to the potential expense of the latter, I've not implemented tests for this explicitly.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #20582

Output from acceptance testing:

```
$ ACM_CERTIFICATE_ROOT_DOMAIN=${TF_DOMAIN} make testacc TESTARGS='-run=TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication_ownership'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGateway*DomainName_MutualTlsAuthentication_ownership -timeout 180m
=== RUN   TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication_ownership
=== PAUSE TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication_ownership
=== CONT  TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication_ownership
--- PASS: TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication_ownership (145.43s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	145.497s

ACM_CERTIFICATE_ROOT_DOMAIN=${TF_DOMAIN} make testacc TESTARGS='-run=TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership -timeout 180m
=== RUN   TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership
=== PAUSE TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership
=== CONT  TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership
--- PASS: TestAccAWSAPIGatewayV2DomainName_MutualTlsAuthentication_ownership (132.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	132.156s
```
